src/main/java/org/broadinstitute/hellbender/tools/examples/ExampleVariantWalker.java

```diff
@@ -0,0 +1,89 @@
+package org.broadinstitute.hellbender.tools.examples;
+
+import htsjdk.variant.variantcontext.VariantContext;
+import org.broadinstitute.hellbender.cmdline.Argument;
+import org.broadinstitute.hellbender.cmdline.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.programgroups.VariantProgramGroup;
+import org.broadinstitute.hellbender.engine.*;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.PrintStream;
+
+/**
+ * Example/toy program that shows how to implement the VariantWalker interface. Prints supplied variants
+ * along with overlapping reads/reference bases/variants (if present).
+ */
+@CommandLineProgramProperties(
+        summary = "Prints variants supplied to the specified output file (stdout if none provided), along with overlapping reads/reference bases/variants (if provided)",
+        oneLineSummary = "Print variants with optional contextual data",
+        programGroup = VariantProgramGroup.class
+)
+public final class ExampleVariantWalker extends VariantWalker {
+
+    @Argument(fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME, shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME, doc = "Output file (if not provided, defaults to STDOUT)", common = false, optional = true)
+    public File outputFile = null;
+
+    @Argument(fullName="auxiliaryVariants", shortName="av", doc="Auxiliary set of variants", optional=true)
+    public FeatureInput<VariantContext> auxiliaryVariants;
```

Comment:
Make `@Argument` fields `private` -- we don't want people to mistakenly think that they have to be `public` in order for the arg injection to work (here and above in `ExampleIntervalWalker`)